fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -0,0 +1,249 @@
+    }
+
+    private StatsDeriveResult computeFilter(Filter filter) {
+        StatsDeriveResult stats = groupExpression.getCopyOfChildStats(0);
+        FilterSelectivityCalculator selectivityCalculator =
+                new FilterSelectivityCalculator(stats.getSlotRefToColumnStatsMap());
+        double selectivity = selectivityCalculator.estimate(filter.getPredicates());
+        stats.multiplyDouble(selectivity);
+        return stats;
+    }
+
+    // TODO: 1. Subtract the pruned partition
+    //       2. Consider the influence of runtime filter
+    //       3. Get NDV and column data size from StatisticManger, StatisticManager doesn't support it now.
+    private StatsDeriveResult computeScan(Scan scan) {
+        Table table = scan.getTable();
+        TableStats tableStats = Utils.execWithReturnVal(() ->
+                Catalog.getCurrentCatalog().getStatisticsManager().getStatistics().getTableStats(table.getId())
+        );
+        Map<Slot, ColumnStats> slotToColumnStats = new HashMap<>();
+        Set<SlotReference> slotSet = scan.getOutput().stream().filter(SlotReference.class::isInstance)
+                .map(s -> (SlotReference) s).collect(Collectors.toSet());
+        for (SlotReference slotReference : slotSet) {
+            String colName = slotReference.getName();
+            if (colName == null) {
+                // TODO: should we throw an exception here?
+                continue;
+            }
+            ColumnStats columnStats = tableStats.getColumnStats(colName);
+            slotToColumnStats.put(slotReference, columnStats);
+        }
+        long rowCount = tableStats.getRowCount();
+        StatsDeriveResult stats = new StatsDeriveResult(rowCount,
+                new HashMap<>(), new HashMap<>());
+        stats.setSlotRefToColumnStatsMap(slotToColumnStats);
+        return stats;
+    }
+
+    private StatsDeriveResult computeAggregate(Aggregate aggregate) {
+        List<Expression> groupByExprList = aggregate.getGroupByExpressionList();
+        StatsDeriveResult childStats = groupExpression.getCopyOfChildStats(0);
+        Map<Slot, ColumnStats> childSlotColumnStatsMap = childStats.getSlotRefToColumnStatsMap();
+        long resultSetCount = -1;
+        for (Expression expression : groupByExprList) {
+            List<SlotReference> slotRefList = expression.collect(SlotReference.class::isInstance);
+            // TODO: Support more complex group expr.
+            //       For example:
+            //              select max(col1+col3) from t1 group by col1+col3;
+            if (slotRefList.size() != 1) {
+                continue;
+            }
+            SlotReference slotRef = slotRefList.get(0);
+            ColumnStats columnStats = childSlotColumnStatsMap.get(slotRef);
+            if (resultSetCount == -1) {
+                resultSetCount = 1;
+                resultSetCount *= columnStats.getNdv();
+            } else if (resultSetCount == 0) {
```

Comment:
Under what circumstances will 0 appear? Does one column with 0 mean that all columns are 0? If not, is there a problem here?